include/devicetree/pm_state.h

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2020 Intel corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DEVICETREE_PM_STATE_H_
+#define ZEPHYR_INCLUDE_DEVICETREE_PM_STATE_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <power/power_state.h>
+
+/**
+ * @defgroup devicetree-pm_states Power Management states device tree macros
+ * @{
+ */
+
+/**
+ * @brief Construct a pm_state_info from 'pm_states' property at index 'i'
+ *
+ * @param node_id A node identifier compatible with pm-states
+ * @param i index of pm_states prop which type is 'phandles'
+ * @return pm_state_info item from 'pm_states' property at index 'i'
+ */
+#define DT_PM_STATE_INFO_ITEM_BY_IDX(node_id, i)                    \
```

Comment:
Do we expect this to work for all architectures (i.e. no variation in state description that's specific to a given platform)?

If so, then I think all this should be able to go into a C file that defines the power state data structures.  No need to have it available for all users.

If not, then it isn't going to work here anyway as it won't support variant interpretations.

Reply:
I'm not sure I understood your question. My goal here is allow one easily extract this info from device tree, I expect application be able work with it in case the app has its own pm policy.